modules/python/nitf/source/nitro.i

```diff
@@ -549,64 +549,70 @@
         /* TODO somehow get the NUMBITSPERPIXEL in the future */
         nitf_Uint8 **buf = NULL;
         PyObject* result = NULL;
-        int i, padded, rowSkip, colSkip, subimageSize;
-        nitf_SubWindow* newWindow = nitf_SubWindow_construct(error);
-        
+        int i, padded, rowSkip, colSkip, subimageSize, bandsMalloced;
```

Comment:
Want to avoid warnings about signed vs. unsigned comparisons.  `i` is used for looping over `window->numBands` which is a `nitf_Uint32`.  If `i` is only used for this, switch it to a `nitf_Uint32`; otherwise, use a different `nitf_Uint32` variable for this.